build_tools/cmake/iree_bitcode_library.cmake

```diff
@@ -227,6 +227,98 @@ function(iree_cuda_bitcode_library)
   )
 endfunction()
 
+# iree_amdgpu_bitcode_library()
+#
+# Builds an AMDGPU LLVM bitcode library from an input file via clang.
+#
+# Parameters:
+# NAME: Name of the target.
+# GPU_ARCH: Target AMDGPU architecture, e.g. gfx942.
+# SRCS: Source files to pass to clang. Headers (*.h) are for dependency
+#       tracking only. Current limitation: only one non-header source is
+#       supported.
+# COPTS: Additional flags to pass to clang.
+# OUT: Output file name. Defaults to {source.c}.{gpu_arch}.bc.
+#
+function(iree_amdgpu_bitcode_library)
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME;OUT;GPU_ARCH"
+    "SRCS;COPTS"
+    ${ARGN}
+  )
+
+  set(_SRC "")
+  foreach(_SRCS_ENTRY IN LISTS _RULE_SRCS)
+    if(_SRCS_ENTRY MATCHES "\.h$")
+      continue()
+    endif()
+    if (_SRC)
+      message(SEND_ERROR "Currently limitation: only one non-header file allowed in SRCS.")
+    endif()
+    set(_SRC "${_SRCS_ENTRY}")
+  endforeach()
+  if(NOT _SRC)
+    message(SEND_ERROR "Error: no non-header file found in SRCS.")
+  endif()
```

Comment:
nit: could log `_RULE_SRCS` in the error messages to help with future debugging

Reply:
good idea